src/Query/Result.php

```diff
@@ -101,4 +101,20 @@ public function aggregations()
     {
         return $this->aggregations;
     }
+
+    /**
+     * @param int $limit the number of hits per request
+     * @return int
+     */
+    public function getNumberOfPages(int $limit): int
+    {
+        if ($limit <= 0) {
+            throw new \InvalidArgumentException("limit parameter must be strictly positive, $limit given");
+        }
+        if (!is_int($this->totalHits)) {
+            throw new \UnexpectedValueException("totalHits seems to be uninitialized, did you call 'getNumberOfPages' on an uninitialised Result ?");
+        }
+        return ceil($this->totalHits / $limit);
+    }
```

Comment:
Maybe we can make it more error-proof by injecting the limit when building the `Result` object.

Reply:
Obviously thought about that, but my problem is that `Result` are created by the `Index`, which doesn't know the queryBuilder, but only the queryBody.

Thinking about it, the queryBody possesses  the limit as a array value, and I can take it from there. Didn't consider that.